dap_aria_mapping/analysis/app/pages/2_Change_Makers.py

```diff
@@ -18,6 +24,190 @@
     page_icon=icon
 )
 
+@st.cache_data(show_spinner = "Loading Data")
+def load_networks(dataset: str) -> nx.Graph:
+    """get the collaboration network for either patents or publications
+
+    Args:
+        dataset (str): "Academia" for publication network or "Industry" for patent network 
+
+    Returns:
+        nx.Graph: collaboration network
+    """
+    return get_collaboration_network(dataset)
+
+@st.cache_data(show_spinner = "Filtering Data to Domain")
+def filter_network_by_domain(domain: str, dataset: str) -> Tuple[nx.Graph, List[str]]:
+    """generates options for the topic filter, and filters the network to only contain nodes of a given domain
+
+    Args:
+        domain (str): domain name from filter
+        dataset (str): either "Academia" or "Industry"
+
+    Returns:
+        Tuple[nx.Graph, List[str]]: subgraph only containing nodes within the domain, and unique areas to populate filter
+    """
+    #must load the network in the function, otherwise won't be able to go back to another domain
+    network = load_networks(dataset)
+    node_domains = nx.get_node_attributes(network, "domains")
+    nodes = [node for node in network.nodes() if domain in node_domains[node]["domain_name"]]
+    return network.subgraph(nodes) 
+
+@st.cache_data(show_spinner = "Filtering Data to Area")
+def filter_network_by_area(area: str, _network: nx.Graph) -> Tuple[nx.Graph, List[str]]:
+    """generates options for the topic filter, and filters the network to only contain nodes of a given domain
+
+    Args:
+        area (str): area name from filter
+        filter_data (pl.DataFrame): lookup table for filtering
+        _network (nx.Graph): collaboration network
+
+    Returns:
+        Tuple[nx.Graph, List[str]]: subgraph only containing nodes within the area, and unique topics to populate filter
+    """
+    node_areas= nx.get_node_attributes(_network, "areas")
+    nodes = [node for node in network.nodes() if area in node_areas[node]["area_name"]]
+    return network.subgraph(nodes)
+
+@st.cache_data
+def filter_network_by_topic(topic: str, _network: nx.Graph) -> nx.Graph:
+    """generates options for the topic filter, and filters the network to only contain nodes of a given domain
+
+    Args:
+        topic (str): topic name from filter
+        _network (nx.Graph): collaboration network
+
+    Returns:
+       nx.Graph : subgraph only containing nodes within the topic
+    """
+    node_topics= nx.get_node_attributes(_network, "topics")
+    nodes = [node for node in network.nodes() if topic in node_topics[node]["topic_name"]]
+    return network.subgraph(nodes)
+
+@st.cache_data
```

Comment:
from what I gather, these should be cache_resource as graph objects are not serializable.